src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    const oldPart = fromFormat[i];
+
+    for (let j = 0; j < splitedDate.length; j++) {
+      const newPart = toFormat[j];
+
+      if (oldPart[0] === newPart[0]) {
+        let newDateElement = splitedDate[i];
+
+        newDateElement = splitedDate[i];
+
+        if (oldPart.length > newPart.length) {
+          newDateElement = newDateElement.split('').slice(-2).join('');
+        }
+
+        if (oldPart.length < newPart.length) {
+          newDateElement =
+            +newDateElement < 30
+              ? `20${newDateElement}`
+              : `19${newDateElement}`;
```

Comment:
When converting from 'YY' to 'YYYY', you are correctly prefixing '19' or '20' based on the year. However, you should ensure that 'newDateElement' is always treated as a string to avoid unexpected type coercion when concatenating the prefix.